src/viur/core/render/json/default.py

```diff
@@ -140,15 +148,38 @@ def renderSkelValues(self, skel: SkeletonInstance, injectDownloadURL: bool = Fal
 
         if (
             injectDownloadURL
-            and (file := getattr(conf.main_app, "file", None))
             and "dlkey" in skel
             and "name" in skel
         ):
-            res["downloadUrl"] = file.create_download_url(
+            res["downloadUrl"] = File.create_download_url(
                 skel["dlkey"],
                 skel["name"],
                 expires=conf.render_json_download_url_expiration
             )
+
+            # generate the downloadUrl for derives
+            search_paths = []
+            if search_path := current.request.get().request.headers.get("X-VIUR-DERIVED-DOWNLOAD-URL"):
+                search_paths.append(search_path)
```

Comment:
If you provide this header multiple times gunicorn merge it comma seperated into one, so we should use split to support it too:


```suggestion
            if search_path := current.request.get().request.headers.get("X-VIUR-DERIVED-DOWNLOAD-URL"):
                search_paths.extend(search_path.split(","))
```